### PR TITLE
Add aot option to gRPC template

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/issues/45888

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/GrpcService-CSharp.csproj.in [hunk 1]
```diff
@@ -4,6 +4,11 @@
     <TargetFramework>${DefaultNetCoreTargetFramework}</TargetFramework>
     <Nullable>enable</Nullable>
     <ImplicitUsings>enable</ImplicitUsings>
+    <!--#if (NativeAot) -->
+    <PublishAot>true</PublishAot>
+    <TrimMode>full</TrimMode>
+    <PublishIISAssets>false</PublishIISAssets>
+    <!--#endif -->
```
Comment (JamesNK) [inline src/ProjectTemplates/Web.ProjectTemplates/GrpcService-CSharp.csproj.in:11]: I copied this from this api template. Can `TrimMode` and `PublishIISAssets` be inferred from `PublishAot`?

Issue to clean up this change in the template: https://github.com/dotnet/aspnetcore/issues/46465

Comment (DamianEdwards) [reply src/ProjectTemplates/Web.ProjectTemplates/GrpcService-CSharp.csproj.in:11]: This has already been cleaned up in the api template as the SDK defaults these now: https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/Web.ProjectTemplates/Api-CSharp.csproj.in#L10-L12

**Comment by JamesNK** [comment]
Tested locally:

image

image

image

**Review by eerhardt — APPROVED** [review]

File: src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs [hunk 2]
```diff
@@ -34,16 +34,42 @@ public ITestOutputHelper Output
...
-    [InlineData(false)]
-    public async Task GrpcTemplate(bool useProgramMain)
+    public async Task GrpcTemplate()
+    {
+        await GrpcTemplateCore();
+    }
+
+    [ConditionalFact(Skip = "Unskip when there are no more build or publish warnings for native AOT.")]
+    [SkipOnHelix("Not supported queues", Queues = "windows.11.arm64.open;" + HelixConstants.Windows10Arm64 + HelixConstants.DebianArm64)]
+    [SkipOnAlpine("https://github.com/grpc/grpc/issues/18338")]
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs:47]: I'll investigate in a follow up PR

**Pull request merged**